netdisco/discoverables/panasonic_viera.py

```diff
@@ -0,0 +1,22 @@
+""" Discovers Panasonic Viera TV devices. """
+
+from . import SSDPDiscoverable
+
+try:
+    from urlparse import urlparse  # Py2
+except ImportError:
+    from urllib.parse import urlparse  # Py3
+
+
+# pylint: disable=too-few-public-methods
+class Discoverable(SSDPDiscoverable):
+    """ Adds support for discovering Viera TV devices. """
+
+    def info_from_entry(self, entry):
+        """ Returns the most important info from a uPnP entry. """
+        return '{}:{}'.format(urlparse(entry.values['location']).hostname,
+                         urlparse(entry.values['location']).port)
+
+    def get_entries(self):
+        """ Get all the Viera TV device uPnP entries. """
+        return self.find_by_st("urn:panasonic-com:service:p00NetworkControl:1")
```

Comment:
Ok 👌

On Tue, Mar 29, 2016, 13:43 Florian Holzapfel <notifications@github.com>
wrote:

> In netdisco/discoverables/panasonic_viera.py
> <https://github.com/balloob/netdisco/pull/18#discussion_r57796761>:
>
> > +except ImportError:
> > +    from urllib.parse import urlparse  # Py3
> > +
> > +
> > +# pylint: disable=too-few-public-methods
> > +class Discoverable(SSDPDiscoverable):
> > +    """ Adds support for discovering Viera TV devices. """
> > +
> > +    def info_from_entry(self, entry):
> > +        """ Returns the most important info from a uPnP entry. """
> > +        return '{}:{}'.format(urlparse(entry.values['location']).hostname,
> > +                         urlparse(entry.values['location']).port)
> > +
> > +    def get_entries(self):
> > +        """ Get all the Viera TV device uPnP entries. """
> > +        return self.find_by_st("urn:panasonic-com:service:p00NetworkControl:1")
>
> After having a look at the Google Play Store listing (
> https://play.google.com/store/apps/details?id=com.panasonic.pavc.viera.vieraremote2&hl=en)
> for the remote app, I think this does only work with Panasonic Viera TVs.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/netdisco/pull/18/files/530534e43b60e7df2815cd1cd93e8f2807bd2c6e#r57796761>
>